### ISSUE TITLE
Lower Accuracy Even with Batch norm.

### ISSUE THREAD

**Issue by learner-codec** [The-AI-Summer/byol-cifar10#2]
Hi thanks for the amazing explanation of the implementation. However I failed to get the reported accuracy reported in the BLOG. With batch Norm I get an accuracy of around 36% which is also visible in the Notebook you have posted. 
<img width="524" alt="image" src="https://user-images.githubusercontent.com/56203705/222877254-d55a379a-47b6-4b4e-8156-6ffeca481560.png">
But on the second block. The accuracy becomes 70% somehow. Please Let me know if I am missing something.

**Comment by dpetrini** [comment]
I run the code in posted notebook and had same issue, Accuracy around 32% (instead of reported around 70%).

Even with MLP BN is set to True as pointed in blog post.

**Comment by dpetrini** [comment]
Hello,

Actually it seems that if you fix incorrect ordering of loss, as commented in issue " Incorrect Specification of loss function. #1 ", it will work like in the blog and will get correct accuracy.
It worked for me.
